test/parallel/test-http2-compat-serverresponse-headers.js

```diff
@@ -102,11 +102,12 @@ server.listen(0, common.mustCall(function() {
     response.setHeader(real, expectedValue);
     const expectedHeaderNames = [real];
     assert.deepStrictEqual(response.getHeaderNames(), expectedHeaderNames);
-    const expectedHeaders = { [real]: expectedValue };
+    const expectedHeaders = Object.create(null);
+    expectedHeaders[real] = expectedValue;
     assert.deepStrictEqual(response.getHeaders(), expectedHeaders);
 
     response.getHeaders()[fake] = fake;
-    assert.strictEqual(response.hasHeader(fake), false);
+    assert.strictEqual(response.hasHeader(fake), true);
```

Comment:
This fails and I would also expect it to fail. If you want to verify that `getHeaders()` has a null prototype that's possible with the following code:

```js
assert.strictEqual(Object.getPrototype(response.getHeaders()), null);
```